pkg/services/ngalert/notifier/images.go

```diff
@@ -20,21 +19,21 @@ func newImageStore(store store.ImageStore) images.ImageStore {
 	}
 }
 
-func (i imageStore) GetImage(ctx context.Context, token string) (*images.Image, error) {
-	image, err := i.store.GetImage(ctx, token)
+// GetImage retrieves an image from the database and checks its URL.
+// If the URL is not a public one, it replaces it with an empty string so that notifiers use the Path field instead.
+func (i *imageStore) GetImage(ctx context.Context, url string) (*images.Image, error) {
+	image, err := i.store.GetImage(ctx, url)
 	if err != nil {
-		if errors.Is(err, models.ErrImageNotFound) {
-			err = images.ErrImageNotFound
-		}
+		return nil, err
 	}
-	var result *images.Image
-	if image != nil {
-		result = &images.Image{
-			Token:     image.Token,
-			Path:      image.Path,
-			URL:       image.URL,
-			CreatedAt: image.CreatedAt,
-		}
+
+	if strings.HasPrefix(image.URL, "file://") {
```

Comment:
we need to explain why we do it here. Please leave a comment that clarifies this.